index.js

```diff
@@ -84,6 +82,30 @@ class K8sVMExecutor extends Executor {
                     throw new Error(`Failed to create pod: ${JSON.stringify(resp.body)}`);
                 }
 
+                return resp.body.metadata.name;
+            })
+            .then(podname => new Promise(resolve => setTimeout(resolve, 5000))
+                .then(() => {
+                    const statusOptions = {
+                        uri: `${this.podsUrl}/${podname}/status`,
+                        method: 'GET',
+                        headers: { Authorization: `Bearer ${this.token}` },
+                        strictSSL: false
+                    };
+
+                    return this.breaker.runCommand(statusOptions);
+                }))
+            .then((resp) => {
+                if (resp.statusCode !== 200) {
+                    throw new Error(`Failed to get pod status: ${JSON.stringify(resp.body)}`);
+                }
+
+                const status = resp.body.status.phase.toLowerCase();
+
+                if (status === 'failed' || status === 'unknown') {
```

Comment:
what if it's still queued ? We need to poll till it's status changes. I have a related requirement https://partner.git.corp.yahoo.com/Screwdriver/engine/issues/920

Reply:
Can probably use `requestretry` pkg for making the status call.
https://github.com/screwdriver-cd/screwdriver/blob/master/features/support/world.js#L6

Reply:
We've done this kind of "wait and poll" in the functional tests

Reply:
We still have to implement `blocked_by`, which will affect the "is still queued" logic.